src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticAnalyzerAttributeAnalyzer.cs

```diff
@@ -133,11 +133,11 @@ protected override void AnalyzeDiagnosticAnalyzer(SymbolAnalysisContext symbolCo
                     // then the analyzer is pretty likely a language-agnostic analyzer.
                     Compilation compilation = symbolContext.Compilation;
                     string compilationTypeNameToCheck = supportsCSharp ? CSharpCompilationFullName : BasicCompilationFullName;
-                    INamedTypeSymbol compilationType = compilation.GetOrCreateTypeByMetadataName(compilationTypeNameToCheck);
+                    INamedTypeSymbol? compilationType = compilation.GetOrCreateTypeByMetadataName(compilationTypeNameToCheck);
                     if (compilationType == null)
                     {
                         string missingLanguage = supportsCSharp ? LanguageNames.VisualBasic : LanguageNames.CSharp;
-                        Diagnostic diagnostic = Diagnostic.Create(AddLanguageSupportToAnalyzerRule, attributeSyntax.GetLocation(), namedType.Name, missingLanguage);
+                        Diagnostic diagnostic = Diagnostic.Create(AddLanguageSupportToAnalyzerRule, attributeSyntax!.GetLocation(), namedType.Name, missingLanguage);
```

Comment:
💡 Add the `RoslynDebug` class to this project (next to NullableAttributes), and convert the assertion above to `RoslynDebug.Assert`. At that point, this suppression will no longer be necessary.

Reply:
Hmm, I already have a Debug.Assert for the same few lines above. Unfortunate why the compiler would not special case Debug.Assert to have semantics equivalent to RoslynDebug.Assert, especially given it will hit almost all the customers. Any case, I have made the change and will also update other places where I had to add suppressions.